### PR TITLE
Support Rewinding while Recording in Sourcereel

### LINKED ISSUE #1149: Sourcereel: Rejecting microphone access then attempt to record + pause crashes the page

### PR DESCRIPTION
**Pull request by jiayushe**
- Rewrote recorder library
- Added logics to support rewinding feature
- Fix #1149

### CODE REVIEW

**Comment by anthony-halim** [comment]
There is huge change on package-lock.json. Have you updated your dependencies to be even with sa_2021?

File: .env.example [hunk 1]
```diff
@@ -8,3 +8,7 @@ REACT_APP_URL_SHORTENER_SIGNATURE=
 REACT_APP_URL_SHORTENER_DOMAIN=
 
 MODULE_BACKEND_URL=http://ec2-54-169-81-133.ap-southeast-1.compute.amazonaws.com
+
+REACT_APP_USER_ROLE=role
+REACT_APP_ACCESS_TOKEN=access_token
+REACT_APP_REFRESH_TOKEN=refresh_token
```
[the remark this answers is not in the document]
Comment (jiayushe) [reply .env.example]: Done.

File: src/components/sourcecast/SourcereelControlbar.tsx [hunk 2]
```diff
@@ -109,37 +117,58 @@ class SourcereelControlbar extends React.PureComponent<
...
+        alert('Microphone not found: ' + error);
+      }
+    );
+  };
+
+  private handleRecorderResuming = () => {
+    const { handleSetEditorReadonly, handleTimerResume } = this.props;
     handleSetEditorReadonly(false);
-    handleTimerStart();
+    handleTimerResume(-1);
```
[the remark this answers is not in the document]
Comment (jiayushe) [reply src/components/sourcecast/SourcereelControlbar.tsx:159]: Negative value means no seeking, resume from the end normally. Added comment for it.

**Comment by jiayushe** [comment]
> Please check the large change to package-lock.json. Is it intended or have you not evened it with sa_2021?

I added my own recorder library so I ran `npm install`, therefore the changes. I have checked the changes, they aren't breaking.

**Pull request merged**